src/Symfony/Component/Form/FormInterface.php

```diff
@@ -101,8 +101,29 @@ public function all();
      * @return FormErrorIterator An iterator over the {@link FormError}
      *                           instances that where added to this form
      */
+    public function getErrorIterator($deep = false, $flatten = true);
+
+    /**
+     * Returns the errors of this form.
+     *
+     * @param bool $deep    Whether to include errors of child forms as well
+     * @param bool $flatten Whether to flatten the list of errors in case
+     *                      $deep is set to true
+     *
+     * @return FormErrorIterator An iterator over the {@link FormError}
+     *                           instances that where added to this form
+     *
+     * @deprecated Use either getErrorIterator or getAllErrors
+     */
     public function getErrors($deep = false, $flatten = true);
 
+    /**
+     * Returns an array of errors for this form.
+     *
+     * @return array|FormError[] An array with all errors found in the form
+     */
+    public function getAllErrors();
```

Comment:
Adding new methods in an interface is a BC break. That cannot be done in a minor version.

Reply:
PR is proposed for 5.0, I guess? So it should be fixed in the description.

Reply:
no. Next version is 4.3, not 5.0. And even then, we would not add a new feature in 5.0 and deprecating the old way at that time. The new API and the deprecation must happen in a 4.x release, as 5.0 will be exactly the same than 4.4, except for the BC layer. Our X.0 releases are free from deprecated APIs.